azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java

```diff
@@ -7,51 +7,82 @@
+    public static final String SUBSCRIPTION_NOT_FOUND = "Subscription %s was not found in current account";
+    public static final String NO_AVAILABLE_SUBSCRIPTION = "No available subscription found in current account";
+
     public static Azure getAzureClient(AuthConfiguration auth, String subscriptionId) throws InvalidConfigurationException, IOException,
         AzureLoginFailureException, InterruptedException, ExecutionException {
-        AzureTokenCredentials azureTokenCredentials = AzureAuthHelper.getAzureTokenCredentials(auth);
+        AzureTokenCredentialsDecorator azureTokenCredentials = AzureAuthHelper.getAzureTokenCredentials(auth);
         if (azureTokenCredentials == null) {
+            AuthType authType;
             final AzureEnvironment environment = AzureEnvironment.AZURE;
             AzureCredential azureCredential;
             try {
+                authType = AuthType.OAUTH;
                 azureCredential = AzureAuthHelper.oAuthLogin(environment);
             } catch (DesktopNotSupportedException e) {
+                authType = AuthType.DEVICE_LOGIN;
                 azureCredential = AzureAuthHelper.deviceLogin(environment);
             }
             AzureAuthHelper.writeAzureCredentials(azureCredential, AzureAuthHelper.getAzureSecretFile());
-            azureTokenCredentials = AzureAuthHelper.getMavenAzureLoginCredentials(azureCredential, environment);
+            azureTokenCredentials = new AzureTokenCredentialsDecorator(authType,
+                    AzureAuthHelper.getMavenAzureLoginCredentials(azureCredential, environment));
         }
 
         if (azureTokenCredentials != null) {
+            // Todo: change to united log
+            Log.info(azureTokenCredentials.getCredentialDescription());
             final Authenticated authenticated = Azure.configure().authenticate(azureTokenCredentials);
             // For cloud shell, use subscription in profile as the default subscription.
             if (StringUtils.isEmpty(subscriptionId) && AzureAuthHelperLegacy.isInCloudShell()) {
                 subscriptionId = AzureAuthHelperLegacy.getSubscriptionOfCloudShell();
             }
-            return StringUtils.isEmpty(subscriptionId) ?
+            final Azure azure = StringUtils.isEmpty(subscriptionId) ?
                     authenticated.withDefaultSubscription() :
-                        authenticated.withSubscription(subscriptionId);
-
+                    authenticated.withSubscription(subscriptionId);
+            checkSubscription(azure, subscriptionId);
+            final Subscription subscription = azure.getCurrentSubscription();
+            Log.info(String.format(SUBSCRIPTION_TEMPLATE, subscription.displayName(), subscription.subscriptionId()));
+            return azure;
         }
 
         return null;
     }
+
+    private static void checkSubscription(Azure azure, String subscriptionId) throws AzureLoginFailureException {
```

Comment:
shall we also support subscription name?

Reply:
if subscriptionId is empty , should return the first one if there are only one

Reply:
Java SDK only support subscription id, and we could only auth with azure first and then get the subscription name.

Reply:
Java SDK will use the first subscription id if no default subscription